src/isomorphic/classic/element/ReactElementValidator.js

```diff
@@ -194,8 +194,9 @@ var ReactElementValidator = {
         false,
         'React.createElement: type should not be null, undefined, boolean, or ' +
           'number. It should be a string (for DOM elements) or a ReactClass ' +
-          '(for composite components).%s',
-        getDeclarationErrorAddendum()
+          '(for composite components).%s%s',
+        getDeclarationErrorAddendum(),
+        ReactComponentTreeHook.getCurrentStackAddendum()
```

Comment:
1. If a debug ID for a mounted/mounting component is available, use that and the stack will be generated by the parent hierarchy.
2. If no debug ID is available (ex: a key/proptypes warning during element creation), we have to rely on the current owner for stack info. We look up the current owner (which is mounted/mounting) and then crawl its parents.
3. The current owner is generally the _parent_ of what we think of as the "current" component/element. So we can also pass the current element (generally the one being constructed) and that gets printed on top.

I think in my ideal world here, we would display `in [unknown] (file.js:123)` when we have the location info (based on the element being constructed – this code can probably be reordered to call ReactElement.createElement first) and leave that line off and just show the parent if we don't (since `in [unknown]` is useless and we won't have any better display name since it isn't a string/function type).